src/commands/Minion/sawmill.ts

```diff
@@ -0,0 +1,121 @@
+import { CommandStore, KlasaMessage } from 'klasa';
+
+import { BotCommand } from '../../lib/BotCommand';
+import { Activity, Tasks, Time } from '../../lib/constants';
+import { hasGracefulEquipped } from '../../lib/gear/functions/hasGracefulEquipped';
+import { Planks } from '../../lib/minions/data/planks';
+import { minionNotBusy, requiresMinion } from '../../lib/minions/decorators';
+import { UserSettings } from '../../lib/settings/types/UserSettings';
+import { SkillsEnum } from '../../lib/skilling/types';
+import { SawmillActivityTaskOptions } from '../../lib/types/minions';
+import { formatDuration, itemNameFromID, stringMatches, toKMB } from '../../lib/util';
+import addSubTaskToActivityTask from '../../lib/util/addSubTaskToActivityTask';
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			altProtection: true,
+			oneAtTime: true,
+			cooldown: 1,
+			usage: '<quantity:int{1}|name:...string> [plankName:...string]',
+			aliases: ['saw'],
+			usageDelim: ' '
+		});
+	}
+
+	@requiresMinion
+	@minionNotBusy
+	async run(msg: KlasaMessage, [quantity, plankName = '']: [null | number | string, string]) {
+		if (typeof quantity === 'string') {
+			plankName = quantity;
+			quantity = null;
+		}
+
+		const plank = Planks.find(
+			plank =>
+				stringMatches(plank.name, plankName) ||
+				stringMatches(plank.name.split(' ')[0], plankName)
+		);
+
+		if (!plank) {
+			return msg.send(
+				`Thats not a valid log to cut. Valid planks are **${Planks.map(
+					plank => plank.name
+				).join(', ')}**.`
+			);
+		}
+
+		const boosts = [];
+		let timePerPlank = (Time.Second * 37) / 27;
+
+		if (hasGracefulEquipped(msg.author.settings.get(UserSettings.Gear.Skilling))) {
+			timePerPlank *= 0.9;
+			boosts.push(`10% for Graceful`);
+		}
+		if (
+			msg.author.skillLevel(SkillsEnum.Woodcutting) >= 60 &&
+			msg.author.settings.get(UserSettings.QP) >= 50
+		) {
+			timePerPlank *= 0.9;
+			boosts.push(`10% for Woodcutting Guild unlocked`);
+		}
+
+		if (quantity === null) {
+			quantity = Math.floor(msg.author.maxTripLength / timePerPlank);
+		}
+
+		if (msg.author.numItemsInBankSync(plank.inputItem) < quantity) {
+			quantity = msg.author.numItemsInBankSync(plank.inputItem);
```

Comment:
put this in a variable instead of reading it twice: `const inputsOwned = msg.author.num...`